doc/Overview/Mvux/Advanced/Pagination.md

```diff
@@ -0,0 +1,395 @@
+
+### Service
+
+Here's the method added to the service (*PeopleService.cs*):
+
+```csharp
+public async ValueTask<int> GetPageCount(uint pageSize, CancellationToken ct) =>
+    (uint)Math.Ceiling(GetPeople().Length / (double)pageSize);
+```
+
+The signature of `GetPageCount` should also be added to the `IPeopleService` interface:
+
+```csharp
+ValueTask<int> GetPageCount(uint pageSize, CancellationToken ct);
+```
+
+### Model
+
+Let's expand the Model with the following:
+
+```csharp
+public partial record PeopleModel(IPeopleService PeopleService)
+{
+    public IListFeed<Person> PeopleAuto ...// will not be used in this example
+
+    const uint PageSize = 20;
+        
+    public IFeed<int> PageCount =>
+        Feed.Async(async (ct) => await PeopleService.GetPageCount(PageSize, ct));
+
+    public IState<uint> CurrentPage => State.Value(this, () => 1u);
+
+    public IListFeed<Person> PeopleManual =>
+        CurrentPage.SelectAsync(async (currentPage, ct) =>
+            // currentPage argument as index based
+            await PeopleService.GetPeopleAsync(pageSize: PageSize, pageIndex: currentPage - 1, ct))
+        .AsListFeed();
+
+    public async ValueTask Move(uint currentPage, uint pageCount, int direction, CancellationToken ct)
+    {
+        var desiredPage = currentPage + direction;
+
+        if (desiredPage <= 0 || desiredPage > pageCount)
+            return;
+
+        await CurrentPage.Set((uint)desiredPage, ct);
+    }
+}
+```
+
+In the code above, whenever `CurrentPage` changes, `PeopleManual` will update its data accordingly.  
+MVUX's code generator generates an async command that is bound to from the View, when the user clicks the designated button to move to either previous or next page, the `Move` method will be invoked with the `direction` argument set to either 1 or -1 added to the current page index.
+
+You may be wondering where the `currentPage` and `pageCount` arguments are populated from, this is another MVUX magic trick where when the parameters type and name matches a Feed or a State in the current Model, their latest value are provided as an argument to the executed Command.  
+Read onward on [Creating commands](xref:Overview.Mvux.Advanced.Commands#creating-commands) to learn more.
+
+### View
+
+Replace the `ListView` from the previous example with this one:
+
+```xml
+<ListView Grid.Column="2" x:Name="manual" ItemsSource="{Binding PeopleManual}"
+          ItemTemplate="{StaticResource PersonDataTemplate}" Header="Load single page on demand">
+    
+    <ListView.Footer>
+        <StackPanel Orientation="Horizontal" Spacing="3">
+
+            <Button Content="Previous" Command="{Binding Move}" VerticalAlignment="Center">
```

Comment:
```suggestion
            <Button Content="Previous" Command="{Binding Move}" VerticalAlignment="Center" CommandParameter="1">
```
This would be much nicer!

Reply:
You bet I've tried that before writing my clunky explicit int declaration 🤢